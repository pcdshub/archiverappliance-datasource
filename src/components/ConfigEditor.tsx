import React, { PureComponent } from 'react';
import { DataSourceHttpSettings, InlineSwitch, InlineField } from '@grafana/ui';
import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
import { AADataSourceOptions } from '../types';

const LABEL_WIDTH = 26;

export type Props = DataSourcePluginOptionsEditorProps<AADataSourceOptions>;

export class ConfigEditor extends PureComponent<Props> {
  constructor(props: Props) {
    super(props);
  }

  onUseBEChange = (event: React.SyntheticEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = this.props;
    const jsonData = {
      ...options.jsonData,
      useBackend: !options.jsonData.useBackend,
    };
    onOptionsChange({ ...options, jsonData });
  };

  render() {
    const { options, onOptionsChange } = this.props;

    return (
      <>
        <DataSourceHttpSettings
          defaultUrl="http://localhost:17668/retrieval"
          dataSourceConfig={options}
          onChange={onOptionsChange}
        />
        <h3 className="page-heading">Misc</h3>
        <div className="gf-form-group">
          <div className="gf-form-inline">
            <InlineField
              label="Use Backend"
              labelWidth={LABEL_WIDTH}
              tooltip="Checking this option will enable the data retrieval with backend. The archived data is retrieved and processed on Grafana server, then the data is sent to Grafana client."
            >
              <InlineSwitch
                value={options.jsonData.useBackend ?? false}
                onChange={this.onUseBEChange}
              />
            </InlineField>
          </div>
        </div>
      </>
    );
  }
}
