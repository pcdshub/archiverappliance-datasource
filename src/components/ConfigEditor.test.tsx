import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ConfigEditor, Props } from './ConfigEditor';

const setup = (propOverrides?: object) => {
  const props: Props = {
    options: {
      id: 1,
      orgId: 1,
      uid: '',
      typeLogoUrl: '',
      name: 'ArchiverAppliance',
      access: 'proxy',
      url: '',
      database: '',
      type: 'ArchiverAppliance',
      typeName: 'ArchiverAppliance',
      user: '',
      basicAuth: false,
      basicAuthUser: '',
      isDefault: true,
      readOnly: false,
      withCredentials: false,
      secureJsonFields: {
        accessKey: false,
        secretKey: false,
      },
      jsonData: {},
      secureJsonData: {
        secretKey: '',
        accessKey: '',
      },
    },
    onOptionsChange: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return shallow(<ConfigEditor {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
