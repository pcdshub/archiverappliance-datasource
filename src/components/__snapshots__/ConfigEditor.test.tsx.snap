// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<Fragment>
  <DataSourceHttpSettings
    dataSourceConfig={
      Object {
        "access": "proxy",
        "basicAuth": false,
        "basicAuthUser": "",
        "database": "",
        "id": 1,
        "isDefault": true,
        "jsonData": Object {},
        "name": "ArchiverAppliance",
        "orgId": 1,
        "readOnly": false,
        "secureJsonData": Object {
          "accessKey": "",
          "secretKey": "",
        },
        "secureJsonFields": Object {
          "accessKey": false,
          "secretKey": false,
        },
        "type": "ArchiverAppliance",
        "typeLogoUrl": "",
        "typeName": "ArchiverAppliance",
        "uid": "",
        "url": "",
        "user": "",
        "withCredentials": false,
      }
    }
    defaultUrl="http://localhost:17668/retrieval"
    onChange={[MockFunction]}
  />
  <h3
    className="page-heading"
  >
    Misc
  </h3>
  <div
    className="gf-form-group"
  >
    <div
      className="gf-form"
    >
      <Switch$1
        checked={false}
        label="Use Backend"
        labelClass="width-13"
        onChange={[Function]}
        tooltip="Checking this option will enable the data retrieval with backend. The archived data is retrieved and processed on Grafana server, then the data is sent to Grafana client."
      />
    </div>
  </div>
</Fragment>
`;
